community[minor]: added a feature to filter documents in Mongoloader

iamKushagra opened the pull request:
"community: added a feature to filter documents in Mongoloader"
    - **Description:** added a feature to filter documents in Mongoloader
    - **Feature:** the feature #18251
    - **Dependencies:** No
    - **Twitter handle:** https://twitter.com/im_Kushagra

iamKushagra committed:
community: added a feature to filter documents by field names in MongodbLoader

iamKushagra committed:
import fix

iamKushagra commented:
@baskaryan

iamKushagra committed:
Merge branch 'master' into MongoLoader_patch

iamKushagra committed:
Merge branch 'master' into MongoLoader_patch

eyurtsev commented on libs/community/langchain_community/document_loaders/mongodb.py:
```diff
@@ -17,6 +17,7 @@ def __init__(
         connection_string: str,
         db_name: str,
         collection_name: str,
+        field_names: Optional[List[str]] = None,
```
Given that this is a named argument could you put it after the `*` and change the type to Optional[Sequence[str]]

iamKushagra replied:
Good point!

iamKushagra replied:
Fixed

eyurtsev commented on libs/community/langchain_community/document_loaders/mongodb.py:
```diff
@@ -66,7 +68,22 @@ async def aload(self) -> List[Document]:
+                    fields = {name: doc[name] for name in self.field_names}
+                except KeyError as err:
+                    logger.warning(f"{err.args[0]} field not found in Mongo document.")
+                    continue  """Skip this document if a specified field is not found"""
```
`"""` is used for doc-strings for functions / classes, not for in code documentation

```suggestion
                    continue  # Skip this document if a specified field is not found
```

seme for changes below

iamKushagra replied:
Fixed

eyurtsev commented on libs/community/langchain_community/document_loaders/mongodb.py:
```diff
@@ -66,7 +68,22 @@ async def aload(self) -> List[Document]:
                 "collection": self.collection_name,
             }
-            result.append(Document(page_content=str(doc), metadata=metadata))
+            if self.field_names is not None:
```
Is there a way to provide the field names as part of the request to mongo instead? So we don't end up sending those fields over the wire if we don't need them.

eyurtsev commented on libs/community/langchain_community/document_loaders/mongodb.py:
```diff
@@ -66,7 +68,22 @@ async def aload(self) -> List[Document]:
+                try:
+                    fields = {name: doc[name] for name in self.field_names}
+                except KeyError as err:
+                    logger.warning(f"{err.args[0]} field not found in Mongo document.")
```
How frequently are fields expected to be missing?

iamKushagra replied:
There could be instances when in a few documents there might that field doesn't exist. So, the frequency is still low but in any such case, we can skip those.

iamKushagra committed:
Updated the code based on PR review.

iamKushagra committed:
Fixed Import

iamKushagra committed:
Include field names in requests to minimize data transfer.

Optimize MongoDB requests: Include field names in requests to minimize data transfer.

iamKushagra committed:
Merge branch 'master' into MongoLoader_patch

iamKushagra committed:
Merge branch 'master' into MongoLoader_patch

eyurtsev commented:
This looks good to me -- if you're able to resolve linting error (`make format`) we can merge

iamKushagra committed:
Fixed linting issue

iamKushagra commented:
Thanks @eyurtsev 🚀

iamKushagra committed:
Fixed Linting Issue

iamKushagra committed:
Merge branch 'langchain-ai:master' into MongoLoader_patch

iamKushagra committed:
Run make format
